scio-core/src/main/scala/com/spotify/scio/coders/ObjectPoolUtils.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2018 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.coders
+
+import org.apache.commons.pool2.impl.{DefaultPooledObject, GenericObjectPool,
+  GenericObjectPoolConfig}
+import org.apache.commons.pool2.{ObjectPool, PooledObject, PooledObjectFactory}
+import scala.util.Try
+
+object ObjectPoolUtils {
+
+
+  def createPool[T](factory: () => T) : ObjectPool[T] = {
+    new GenericObjectPool[T](new PooledObjectFactory[T] {
```

Comment:
it seems that kryo was it's own pooling support. what do you think of giving it a try first and introduce commons-pool if doesn't work out? 

https://github.com/EsotericSoftware/kryo#pooling-kryo-instances

They provide an example and i think it's pretty straightforward to replace commons-pool.